pkg/controller/hawtio/hawtio_controller.go

```diff
@@ -718,3 +735,76 @@ func (r *ReconcileHawtio) deletion(ctx context.Context, hawtio *hawtiov1alpha1.H
 
 	return nil
 }
+
+func generateCertificateSecret(name string, secretWithCa *corev1.Secret, log logr.Logger) (*corev1.Secret, error) {
+	caCertFile := secretWithCa.Data["tls.crt"]
+	pemBlock, _ := pem.Decode(caCertFile)
+	if pemBlock == nil {
+		return nil, errors2.New("failed to decode certificate")
+	}
+	caCert, err := x509.ParseCertificate(pemBlock.Bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	caKey := secretWithCa.Data["tls.key"]
+	pemBlock, _ = pem.Decode(caKey)
+	if pemBlock == nil {
+		return nil, errors2.New("failed to decode certificate key")
+	}
+	caPrivateKey, err := x509.ParsePKCS1PrivateKey(pemBlock.Bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	subj := pkix.Name{
+		CommonName: "fuse-console.fuse.svc",
```

Comment:
I'd suggest to expose a parameter into the `Hawtio.Spec` struct so that the common name can be configured.

Also the default should be changed to `hawtio-online.hawtio.svc` for upstream, and a build-time variable should be added so that the default can be changed by productisation builds.